### PR TITLE
Feature/#1506 forced batch

### LINKED ISSUE #1506: Review forceBatch

### PR DESCRIPTION
**Pull request by ARR552**
Closes #1506 

### What does this PR do?

Adds unit test in the synchronicer and applies some fixes

### Reviewers

- @cool-develope 
- @tclemos 
- @arnaubennassar

### CODE REVIEW

File: state/pgstatestorage.go [hunk 1]
```diff
@@ -112,21 +109,34 @@ func (p *PostgresStorage) getExecQuerier(dbTx pgx.Tx) execQuerier {
 // Reset resets the state to a block for the given DB tx
 func (p *PostgresStorage) Reset(ctx context.Context, blockNumber uint64, dbTx pgx.Tx) error {
 	e := p.getExecQuerier(dbTx)
+	const resetForcedBatchSQL = `
+		UPDATE state.forced_batch 
+		SET batch_num = null
+		FROM state.virtual_batch 
+		INNER JOIN state.block ON state.block.block_num = state.virtual_batch.block_num
```
Comment (cool-develope) [inline state/pgstatestorage.go:116]: why we need the `inner join` with block table? `virtual_batch` has already `block_num` ???

File: state/pgstatestorage.go [hunk 2]
```diff
@@ -112,21 +109,34 @@ func (p *PostgresStorage) getExecQuerier(dbTx pgx.Tx) execQuerier {
 // Reset resets the state to a block for the given DB tx
 func (p *PostgresStorage) Reset(ctx context.Context, blockNumber uint64, dbTx pgx.Tx) error {
```
Comment (cool-develope) [inline state/pgstatestorage.go:110]: what is the plan for the test of the `Reset` function?

File: sequencer/broadcast/server.go [hunk 3]
```diff
@@ -119,7 +115,7 @@ func (s *Server) genericGetBatch(ctx context.Context, batch *state.Batch) (*pb.G
 		RollupExitRoot:    rollupExitRoot,
 		Timestamp:         uint64(batch.Timestamp.Unix()),
 		Transactions:      transactions,
-		ForcedBatchNumber: forcedBatchNum,
+		ForcedBatchNumber: fb,
```
Comment (arnaubennassar) [inline sequencer/broadcast/server.go:118]: maybe this should b a pointer, as I guess that the first `ForcedBatchNumber` will be 0

**Review by arnaubennassar — CHANGES_REQUESTED** [review]
If I understood the code well, if there is an inconsistency on the DB related to the forced batches, the synchronizer will return error, and potentially enter and endless loop of errors.

I think this is good enough for now, but eventually the behaviour should be more like the trusted state: if the content of the DB is unexpected, we log an error, but we put whatever it's on the virtual state into the node's DB

File: sequencer/broadcast/server.go (hunk 3 above)
Comment (ARR552) [reply sequencer/broadcast/server.go:118]: No, In this case the first forcedBatch will be one

**Comment by ARR552** [comment]
> If I understood the code well, if there is an inconsistency on the DB related to the forced batches, the synchronizer will return error, and potentially enter and endless loop of errors.
> 
> I think this is good enough for now, but eventually the behaviour should be more like the trusted state: if the content of the DB is unexpected, we log an error, but we put whatever it's on the virtual state into the node's DB

Yes, I think that for initial steps of this feature testing it is good to detect the problem as soon as possible. We can modify it anytime.

**Review by ToniRamirezM — APPROVED** [review]

**Review by arnaubennassar — APPROVED** [review]

**Pull request merged**